### PR TITLE
Add lots of help messages

### PR DESCRIPTION
**Pull request by paulcsmith**
Also makes it so gen.migration prints a help message

Part of https://github.com/luckyframework/lucky/issues/787

### CODE REVIEW

File: src/precompiled_tasks/gen/migration.cr [hunk 1]
```diff
@@ -1,3 +1,3 @@
 require "../../avram"
 
-Gen::Migration.new.call
+Gen::Migration.new.print_help_or_call(ARGV)
```
Comment (paulcsmith) [inline src/precompiled_tasks/gen/migration.cr]: This makes it so the task prints the help message if given `-h`, `help` or `--help`

File: src/avram/tasks/db/redo.cr [hunk 2]
```diff
@@ -3,6 +3,17 @@ require "colorize"
 class Db::Redo < LuckyCli::Task
   summary "Rollback then run the last migration"
 
+  def help_message
+    <<-TEXT
+    Rollback then run the last migration.
```
Comment (jwoertink) [inline src/avram/tasks/db/redo.cr]: Rollback how many? Just the last one?

File: src/avram/tasks/db/rollback_to.cr [hunk 3]
```diff
@@ -3,6 +3,19 @@ require "colorize"
...
 
+  def help_message
+    <<-TEXT
+    Rollback to a specific migration.
+
+    You can get the migration version from the filename or by running 'lucky db.migration.status'
+
+    Example:
+
+      lucky db.rollback_to 20180802180356
```
Comment (jwoertink) [inline src/avram/tasks/db/rollback_to.cr]: Examples like this are what really sell this. Love it!

File: src/avram/tasks/db/schema_dump.cr [hunk 4]
```diff
@@ -4,6 +4,19 @@ class Db::Schema::Dump < LuckyCli::Task
   def initialize(@dump_to : String? = nil, @quiet : Bool = false)
   end
 
+  def help_message
+    <<-TEXT
+    Exports/dumps the database schema to a sql file.
```
Comment (jwoertink) [inline src/avram/tasks/db/schema_dump.cr]: I like this, but I wonder if it should be noted that this doesn't include the data in the tables? Also, should that be an option? 🤔

File: src/avram/tasks/db/schema_restore.cr [hunk 5]
```diff
@@ -4,6 +4,19 @@ class Db::Schema::Restore < LuckyCli::Task
   def initialize(@import_file_path : String? = nil, @quiet : Bool = false)
   end
 
+  def help_message
+    <<-TEXT
+    Restores the database from a sql dump file.
+
+    You must specify the path that you want Avram to dump the sql to.
```
Comment (jwoertink) [inline src/avram/tasks/db/schema_restore.cr]:
```suggestion
    You must specify the path to the sql that you want Avram to restore from.
```

File: src/avram/tasks/db/schema_dump.cr [hunk 6]
```diff
@@ -4,6 +4,19 @@ class Db::Schema::Dump < LuckyCli::Task
...
+
+    You must specify the path that you want Avram to dump the sql to.
+
+    Example:
+
+      lucky db.dump structure.sql
```
Comment (jwoertink) [inline src/avram/tasks/db/schema_dump.cr]:
```suggestion
      lucky db.schema.dump structure.sql
```

File: src/avram/tasks/db/schema_restore.cr [hunk 7]
```diff
@@ -4,6 +4,19 @@ class Db::Schema::Restore < LuckyCli::Task
...
+
+    Example:
+
+      lucky db.restore structure.sql
```
Comment (jwoertink) [inline src/avram/tasks/db/schema_restore.cr]:
```suggestion
      lucky db.schema.restore structure.sql
```

File: src/avram/tasks/db/redo.cr (hunk 2 above)
Comment (paulcsmith) [reply src/avram/tasks/db/redo.cr]: Oh good point

File: src/avram/tasks/db/schema_dump.cr (hunk 4 above)
Comment (paulcsmith) [reply src/avram/tasks/db/schema_dump.cr]: Good point. And I don't think so on exporting data

File: src/avram/tasks/db/schema_restore.cr (hunk 5 above)
Comment (paulcsmith) [reply src/avram/tasks/db/schema_restore.cr]: Much better

**Pull request merged**